### PR TITLE
jv_homework_fruit-shop-tests

### PR DESCRIPTION
**Pull request by etezaadi**
added tests for fruit-shop

### CODE REVIEW

File: src/test/java/core/basesyntax/dao/FruitShopDaoImplTest.java [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+public class FruitShopDaoImplTest {
+    private static FruitShopDao fruitShopDao;
+
+    @BeforeClass
+    public static void beforeClass() {
+        fruitShopDao = new FruitShopDaoImpl();
+    }
+
+    @Test
+    public void fruitShopDao_add_isOk() {
```
Comment (MaliukDaria) [inline src/test/java/core/basesyntax/dao/FruitShopDaoImplTest.java:18]:
```suggestion
    public void fruitShopDao_add_Ok() {
```

File: src/test/java/core/basesyntax/dao/FruitShopDaoImplTest.java [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+        fruitShopDao.add("apple", 100);
+        fruitShopDao.add("banana", 200);
+        int expected = 2;
+        int actual = Storage.fruitStorage.size();
+        Assert.assertEquals(expected, actual);
+    }
+
+    @Test
+    public void fruitShopDao_getAll_isOk() {
```
Comment (MaliukDaria) [inline src/test/java/core/basesyntax/dao/FruitShopDaoImplTest.java:27]:
```suggestion
    public void fruitShopDao_getAll_Ok() {
```

fix in all places

File: src/test/java/core/basesyntax/dao/FruitShopDaoImplTest.java [hunk 3]
```diff
@@ -0,0 +1,40 @@
+package core.basesyntax.dao;
+
+import core.basesyntax.db.Storage;
+import org.junit.After;
+import org.junit.Assert;
+import org.junit.BeforeClass;
+import org.junit.Test;
+
+public class FruitShopDaoImplTest {
+    private static FruitShopDao fruitShopDao;
```
Comment (MaliukDaria) [inline src/test/java/core/basesyntax/dao/FruitShopDaoImplTest.java:10]: Lets add test for getAll method while Storage is empty

File: src/test/java/core/basesyntax/service/impl/CsvFileReaderServiceImplTest.java [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+                                        "r,apple,10",
+                                        "p,apple,20",
+                                        "p,banana,5",
+                                        "s,banana,50");
+        List<String> actual = csvFileReaderService.readFromFile(VALID_FILE);
+        assertEquals(expected, actual);
+    }
+
+    @Test
+    public void readFromEmptyFile() {
```
Comment (MaliukDaria) [inline src/test/java/core/basesyntax/service/impl/CsvFileReaderServiceImplTest.java:36]:
```suggestion
    public void readFromEmptyFile_Ok() {
```

File: src/test/java/core/basesyntax/service/impl/CsvFileWriterServiceImplTest.java [hunk 5]
```diff
@@ -0,0 +1,56 @@
...
+        csvFileWriterService.writeToFile(WRITE_TO_FILE, reportService);
+        String actual = readFromFile();
+        String expected = "fruit,quantity" + System.lineSeparator()
+                + "banana," + 135 + System.lineSeparator()
+                + "apple," + 35 + System.lineSeparator();
+        Assert.assertEquals(expected, actual);
+    }
+
+    @Test(expected = RuntimeException.class)
+    public void writeToFile_isNotOk() {
```
Comment (MaliukDaria) [inline src/test/java/core/basesyntax/service/impl/CsvFileWriterServiceImplTest.java:40]:
```suggestion
    public void writeToFile_invalidPath_NotOk() {
```

File: src/test/java/core/basesyntax/service/impl/CsvFileWriterServiceImplTest.java [hunk 6]
```diff
@@ -0,0 +1,56 @@
...
+import core.basesyntax.service.ReportService;
+import java.io.File;
+import java.io.IOException;
+import java.nio.file.Files;
+import org.junit.After;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Test;
+
+public class CsvFileWriterServiceImplTest {
```
Comment (MaliukDaria) [inline src/test/java/core/basesyntax/service/impl/CsvFileWriterServiceImplTest.java:15]: leta add test to write to null path

File: src/test/java/core/basesyntax/service/impl/FruitTransactionServiceImplTest.java [hunk 7]
```diff
@@ -0,0 +1,65 @@
...
+                new PurchaseOperationHandler());
+        operationHandlerHashMap.put(Operations.RETURN.getOperation(),
+                new ReturnOperationHandler());
+    }
+
+    @Before
+    public void setUp() {
+        operationStrategy = new OperationStrategyImpl(operationHandlerHashMap);
+        fruitTransactionService = new FruitTransactionServiceImpl(operationStrategy);
+        fruitShopDao = new FruitShopDaoImpl();
```
Comment (MaliukDaria) [inline src/test/java/core/basesyntax/service/impl/FruitTransactionServiceImplTest.java:46]: I think we can do it in @BeforeClass method

File: src/test/java/core/basesyntax/service/impl/CsvFileWriterServiceImplTest.java [hunk 8]
```diff
@@ -0,0 +1,56 @@
...
+    private static final String WRITE_TO_FILE = "src/main/resources/after.csv";
+    private static final String INCORRECT_PATH = "incorrect/path";
+    private static CsvFileWriterService csvFileWriterService;
+    private static ReportService reportService;
+
+    @Before
```
Comment (MaliukDaria) [inline src/test/java/core/basesyntax/service/impl/CsvFileWriterServiceImplTest.java:21]:
```suggestion
    @BeforeClass
```

fix in all places

File: src/test/java/core/basesyntax/service/impl/FruitTransactionServiceImplTest.java [hunk 9]
```diff
@@ -0,0 +1,65 @@
...
+    }
+
+    @Test
+    public void fruitTransactionService_isOk() {
```
Comment (MaliukDaria) [inline src/test/java/core/basesyntax/service/impl/FruitTransactionServiceImplTest.java:50]:
```suggestion
    public void fruitTransactionService_processTransactions_isOk() {
```

File: src/test/java/core/basesyntax/service/impl/ReportServiceImplTest.java [hunk 10]
```diff
@@ -0,0 +1,40 @@
...
+import core.basesyntax.dao.FruitShopDao;
+import core.basesyntax.dao.FruitShopDaoImpl;
+import core.basesyntax.db.Storage;
+import core.basesyntax.service.ReportService;
+import org.junit.After;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Test;
+
+public class ReportServiceImplTest {
```
Comment (MaliukDaria) [inline src/test/java/core/basesyntax/service/impl/ReportServiceImplTest.java:12]: lets add test for creating report for empty Storage

File: src/test/java/core/basesyntax/service/strategy/impl/BalanceOperationHandlerTest.java [hunk 11]
```diff
@@ -0,0 +1,31 @@
...
+    private static OperationHandler balanceOperationHandler;
+
+    @Before
+    public void setUp() {
+        balanceOperationHandler = new BalanceOperationHandler();
+    }
+
+    @Test
+    public void balanceOperationHandler_isOk() {
+        Storage.fruitStorage.put("apple", null);
```
Comment (MaliukDaria) [inline src/test/java/core/basesyntax/service/strategy/impl/BalanceOperationHandlerTest.java:20]: Why do we have put null quantity here?

File: src/test/java/core/basesyntax/service/strategy/impl/OperationStrategyImplTest.java [hunk 12]
```diff
@@ -0,0 +1,61 @@
...
+import core.basesyntax.service.strategy.OperationHandler;
+import core.basesyntax.service.strategy.OperationStrategy;
+import java.util.HashMap;
+import java.util.Map;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.BeforeClass;
+import org.junit.Test;
+
+public class OperationStrategyImplTest {
```
Comment (MaliukDaria) [inline src/test/java/core/basesyntax/service/strategy/impl/OperationStrategyImplTest.java:13]: lets add some test for unsupported operation

**Review by MaliukDaria — CHANGES_REQUESTED** [review]

**Comment by etezaadi** [comment]
@MaliukDaria thanks for your support! Already fixed :)

File: src/test/java/core/basesyntax/service/impl/ReportServiceImplTest.java [hunk 13]
```diff
@@ -0,0 +1,46 @@
...
+                + "banana," + 20 + System.lineSeparator()
+                + "orange," + 30 + System.lineSeparator()
+                + "apple," + 10 + System.lineSeparator();
+        Assert.assertEquals(expected, actual);
+    }
+
+    @Test
+    public void reportService_createForEmptyStorage_Ok() {
+        boolean empty = Storage.fruitStorage.isEmpty();
+        Assert.assertTrue(empty);
```
Comment (MaliukDaria) [inline src/test/java/core/basesyntax/service/impl/ReportServiceImplTest.java:39]:
```suggestion
        String actual = reportService.createReport();
        compare actual and expected here
```

File: src/test/java/core/basesyntax/service/strategy/impl/OperationStrategyImplTest.java [hunk 14]
```diff
@@ -0,0 +1,79 @@
...
+
+    @Test
+    public void returnOperationHandler_Ok() {
+        OperationHandler returnOperation = operationStrategy.get(Operations.RETURN.getOperation());
+        Assert.assertTrue(operationHandlerMap.containsValue(returnOperation));
+    }
+
+    @Test
+    public void operationStrategy_invalidOperation_NotOk() {
+        operationStrategy.get(INVALID_OPERATION_CHARACTER);
```
Comment (MaliukDaria) [inline src/test/java/core/basesyntax/service/strategy/impl/OperationStrategyImplTest.java:67]: What do we expect to see for unsupported operation? (I think it should be some exception)

**Review by MaliukDaria — CHANGES_REQUESTED** [review]

File: src/test/java/core/basesyntax/service/strategy/impl/OperationStrategyImplTest.java [hunk 15]
```diff
@@ -0,0 +1,80 @@
...
+    }
+
+    @Test (expected = RuntimeException.class)
+    public void operationStrategy_invalidOperation_NotOk() {
+        operationStrategy.get(INVALID_OPERATION_CHARACTER)
+                .handle("apple", 100);
+    }
+
+    @Test
+    public void operationStrategy_operationIsEmpty_NotOk() {
```
Comment (MaliukDaria) [inline src/test/java/core/basesyntax/service/strategy/impl/OperationStrategyImplTest.java:72]: notOk it's when we expect some exception, and here we just pass null and nothing happened

**Review by MaliukDaria — APPROVED** [review]

**Pull request closed without merging**